src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java

```diff
@@ -311,20 +320,47 @@ else if (SiteUtil.JAVADOC_TOKENS.equals(propertyName)) {
         }
         else {
             final String type = SiteUtil.getType(field, propertyName, currentModuleName, instance);
-            final String relativePathToPropertyTypes =
-                    SiteUtil.getLinkToDocument(currentModuleName, "property_types.xml");
-            final String escapedType = type
-                    .replace("[", ".5B")
-                    .replace("]", ".5D");
-            final String url =
-                    String.format(Locale.ROOT, "%s#%s", relativePathToPropertyTypes, escapedType);
-            sink.link(url);
-            sink.text(type);
-            sink.link_();
+            if (PropertyType.TOKEN_ARRAY.getDescription().equals(type)) {
+                processLinkForTokenTypes(sink);
+            }
+            else {
+                final String relativePathToPropertyTypes =
+                        SiteUtil.getLinkToDocument(currentModuleName, PROPERTY_TYPES_XML);
+
+                final String encodedType = URLEncoder.encode(type, StandardCharsets.UTF_8);
+
+                final String encTypeF = encodedType
+                        .replace("%5B", ".5B")
+                        .replace("%5D", ".5D");
```

Comment:
I was under the impression these were not needed when we use the URL encoder. Is something wrong?

Reply:
If i just use the encoder, it alters the url for the following files in this manner
```
src/xdocs/checks/coding/illegalcatch.xml
src/xdocs/checks/coding/illegalinstantiation.xml 
src/xdocs/checks/coding/illegalthrows.xml 
src/xdocs/checks/whitespace/filetabcharacter.xml
```
```
-              <td><a href="../../property_types.html#String.5B.5D">String[]</a></td>
+              <td><a href="../../property_types.html#String%5B%5D">String[]</a></td>
```

Reply:
ok, that is interesting. Doxia is basically creating an invalid link drop as it is using `.` instead of `%`. This is why the URL encoder uses `%`.

@romani I question, should we still be using URL encoder?

Reply:
@relentless-pursuit , can we avoid `encodedType = URLEncoder.encode`?

Reply:
Yes, i will restore it back to original.

Reply:
> @relentless-pursuit , can we avoid `encodedType = URLEncoder.encode`?

done